Source/Core/Common/Debug/CodeTrace.cpp

```diff
@@ -0,0 +1,394 @@
+// Copyright 2022 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Common/Debug/CodeTrace.h"
+
+#include <chrono>
+#include <regex>
+
+#include "Common/Event.h"
+#include "Common/StringUtil.h"
+#include "Core/Debugger/PPCDebugInterface.h"
+#include "Core/HW/CPU.h"
+#include "Core/PowerPC/PowerPC.h"
+
+static bool CompareInstruction(std::string_view instruction, std::vector<std::string> type_compare)
+{
+  for (auto& s : type_compare)
+  {
+    if (StringBeginsWith(instruction, s))
+      return true;
+  }
+
+  return false;
+}
+
+static bool IsInstructionLoadStore(std::string_view ins)
+{
+  return (StringBeginsWith(ins, "l") && !StringBeginsWith(ins, "li")) ||
+         StringBeginsWith(ins, "st") || StringBeginsWith(ins, "psq_l") ||
+         StringBeginsWith(ins, "psq_s");
+}
+
+const u32 GetMemoryTargetSize(const std::string& instr)
+{
+  // Word-size operations are taken as the default, check the others.
+  u32 size = 4;
+
+  const std::string op = instr.substr(0, 4);
```

Comment:
Could the input be a `string_view` then this wouldn't need to allocate a ne string?